scripts/mkdist.js

```diff
@@ -1,30 +1,30 @@
-var fs = require('fs')
+const fs = require('fs')
+const {join} = require('path')
+const mkdirp = require('mkdirp')
```

Comment:
@Dignifiedquire please don't use this, this forces users to have all the ES2016 harmony stuff just to build the `go-ipfs` dist. ES2016 it's widely deployed yet. this forces installations to match your own, and increases the complexity of messing with this repo.

Across the project, we take "env dependencies" and any "dependencies that need to be installed" very seriously. We don't even use `bash` in _lots_ of places in our scrips and code to ensure that things are more cross-platform and work for users on the first try. It's not easy to do this but it yields much smoother experiences for the large diversity of platforms and dev environments touching our code.

Also, though this `mkdist.js` stuff requires node already, it was (and still is) my intention to remove this dependency and move to shell, or a self-contained go-binary that is installed via `gx` or `ipget`, called by `make`. depending on the users having `node` is a _big_ expectation. depending on them having the shiniest newest babelified harmonified sprinkled with ponies version, is a _huge_ expectation.

i know this is time consuming, and that npm solved so much of this. but most of our repos operate _outside_ of the nice npm bubble. which is why we made `gx` and eventually plan to use it and install it in these repos when necessary.